clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -109,21 +112,31 @@ public void configure(Map<String, ?> configs) throws KafkaException {
 
     @Override
     public void validateReconfiguration(Map<String, ?> newConfigs) {
-        createNewSslEngineFactory(newConfigs);
+        Utils.closeQuietly(mayCreateNewSslEngineFactory(newConfigs), "close temporary ssl engine factory");
```

Comment:
It seems to me that creating an new ssl engine factory to validate reconfiguration is a bit weird but it should be addressed by another issue.